### PR TITLE
Iox #1196 fix list warnings

### PR DESCRIPTION
**Pull request by elfenpiff**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
    - [x] Each unit test case has a unique UUID
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Part of #1196

### CODE REVIEW

File: iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/list.inl [hunk 1]
```diff
@@ -26,19 +26,28 @@ namespace iox
 {
 namespace cxx
 {
+/// m_data fields are explicitly initialized whenever a new element is inserted
+/// into the list
+/// @NOLINTNEXTLINE(cppcoreguidelines-pro-type-member-init,hicpp-member-init)
```
Comment (elBoberido) [inline iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/list.inl:31]: I'm not sure what happens when there is a doxygen comment without a doxygen tag. I would suggest to always use `@NOLINTJUSTIFICATION` with doxygen comments

File: iceoryx_hoofs/include/iceoryx_hoofs/cxx/list.hpp [hunk 2]
```diff
@@ -379,9 +382,13 @@ class list
     // to the beginning and end of the list. This additional element (index position 'capacity' aka
     // BEGIN_END_LINK_INDEX) 'previous' will point to the last valid element (end()) and 'next' will point to the
     // first used list element (begin())
+
+    /// @NOLINTJUSTIFICATION will be replaced by uninitialized array
```
Comment (FerdinandSpitzschnueffler) [inline iceoryx_hoofs/include/iceoryx_hoofs/cxx/list.hpp:386]: Please add a todo #1196

File: iceoryx_hoofs/test/moduletests/test_cxx_list.cpp [hunk 3]
```diff
@@ -2394,7 +2441,9 @@ TEST_F(list_test, ListIsCopyableViaMemcpy)
             sut1.emplace_front(static_cast<int64_t>(j));
         }
 
-        memcpy(reinterpret_cast<void*>(otherSutPtr), reinterpret_cast<const void*>(&sut1), sizeof(sut1));
+        /// @NOLINTJUSTIFICATION actually the list is trivially copyable and this test verifies this with memcpy
```
Comment (FerdinandSpitzschnueffler) [inline iceoryx_hoofs/test/moduletests/test_cxx_list.cpp:2444]:
```suggestion
        /// @NOLINTJUSTIFICATION the list is trivially copyable and this test verifies this with memcpy
```

File: iceoryx_hoofs/test/moduletests/test_cxx_list.cpp [hunk 4]
```diff
@@ -49,6 +49,7 @@ class list_test : public Test
             classValue = m_value;
         }
 
+        /// @NOLINTNEXTLINE(hicpp-explicit-conversions)
```
Comment (FerdinandSpitzschnueffler) [inline iceoryx_hoofs/test/moduletests/test_cxx_list.cpp:52]:
```suggestion
        /// @NOLINTNEXTLINE(hicpp-explicit-conversions) only used in test
```

**Pull request merged**